include/sys/device.h

```diff
@@ -29,13 +29,23 @@ typedef enum {
   DIF_COUNT /* this must be the last item */
 } drv_if_t;
 
+/* During boot, the device tree is scanned multiple times.
+ * Each scan, drivers may be attached to devices. Each driver is assigned
+ * a pass number. Drivers may only probe and attach to devices if their
+ * pass numer is equal to the current pass number. */
+typedef enum {
+  FIRST_PASS,
+  SECOND_PASS,
+} pass_num_t;
+
 struct driver {
   const char *desc;            /* short driver description */
   size_t size;                 /* device->state object size */
   d_probe_t probe;             /* probe for specific device(s) */
   d_attach_t attach;           /* attach device to system */
   d_detach_t detach;           /* detach device from system */
   void *interfaces[DIF_COUNT]; /* pointers to device methods structures */
+  pass_num_t pass;             /* device tree pass number */
```

Comment:
Could you move it just after `size` - here and in every definition?

Reply:
Done.